### PR TITLE
Add content type in upload s3

### PR DESCRIPTION
**Pull request by eduardoscottini**

### CODE REVIEW

File: lib/ecto/upload.ex [hunk 1]
```diff
@@ -17,19 +17,21 @@ defmodule Uploadex.Upload do
...
-      {:ok, binary} ->
+      {content_type, {:ok, binary}} ->
         {:ok, %{
           filename: generate_filename(filename),
-          binary: binary
+          binary: binary,
+          content_type: content_type
         }}
 
       error ->
```
Comment (gabrielpra1) [inline lib/ecto/upload.ex]: we should probably handle `{:error, _}` and `{_, {:error, _}`

File: lib/processing/file_processing.ex [hunk 2]
```diff
@@ -6,11 +6,11 @@ defmodule Uploadex.FileProcessing do
   @doc """
   If it's in base64, decode it. Otherwise, do not try to process the file.
   """
-  @spec process_binary(String.t) :: {:ok, binary()} | :error | {:error, keyword()}
+  @spec process_binary(String.t) :: {String.t, {:ok, binary()}} | :error | {:error, keyword() | String.t}
```
Comment (gabrielpra1) [inline lib/processing/file_processing.ex]: Should be

```
@spec process_binary(String.t) :: {String.t, {:ok, binary()}} | {:error, String.t} | {String.t, {:error, keyword}}
```

Or maybe handle this to always return `{:error, keyword | String.t}`?

`{:ok, %{content_type: "", binary: <<>>}` and `{:error, error}` would be a nicer API

File: lib/storage/s3_storage.ex [hunk 3]
```diff
@@ -23,10 +23,13 @@ defmodule Uploadex.S3Storage do
...
   @impl true
-  def store(%{filename: filename, path: path}, opts) do
+  def store(%{filename: filename, path: path, content_type: content_type}, opts) do
     bucket = Keyword.fetch!(opts, :bucket)
     directory = Keyword.fetch!(opts, :directory)
-    upload_opts = Keyword.get(opts, :upload_opts, [])
+    upload_opts =
+      opts
+      |> Keyword.get(:upload_opts, [])
+      |> Keyword.put_new(:content_type, content_type)
```
Comment (gabrielpra1) [inline lib/storage/s3_storage.ex:33]: Please document that we can pass `content_type` as upload_opts for S3 storage, and that the default is the upload's content type

File: lib/validation.ex [hunk 4]
```diff
@@ -26,6 +26,8 @@ defmodule Uploadex.Validation do
     end
   end
 
+  defp all_extensions_accepted?(_files, :any), do: true
```
Comment (gabrielpra1) [inline lib/validation.ex:29]: Is this necessary? We are already handling this in `def validate_extensions(_files, :any), do: :ok`

**Review by gabrielpra1 — CHANGES_REQUESTED** [review]

File: lib/validation.ex (hunk 4 above)
Comment (eduardoscottini) [reply lib/validation.ex:29]: Yes, because of the last breaking change in the lib

File: lib/processing/file_processing.ex [hunk 5]
```diff
@@ -6,11 +6,18 @@ defmodule Uploadex.FileProcessing do
...
   def process_binary(image_binary) do
-    case String.split(image_binary, ";base64,") do
-      [_metadata, base64] -> Base.decode64(base64)
-      _ -> {:ok, image_binary}
+    with  [metadata, base64] <- String.split(image_binary, ";base64,"),
+          {:ok, binary} <- Base.decode64(base64)
+    do
+      {:ok, %{binary: binary, content_type: String.replace(metadata, "data:", "")}}
+    else
+      [_base64] -> {:error, "Invalid base64 format"}
```
Comment (gabrielpra1) [inline lib/processing/file_processing.ex]: I think we should move this `"Invalid base64 format"` to a module attribute

**Review by gabrielpra1 — CHANGES_REQUESTED** [review]

**Pull request merged**